### DISCUSSION TITLE
Adding tests

### DISCUSSION THREAD

**Discussion by Avi98** [Fredkiss3/gh-next#42] [General]
I can create an issue and start adding unit tests for components using jest.

We can also add E2E tests for page level

**Comment by Fredkiss3** [comment]
Is it possible to use something like vitest for example as jest can be slow.

Though I think Jest doesn't work correctly with server actions (files with the `use server` directive at the top) 🤔 ?
My logic is mostly handled within server actions (files inside `src/app/(actions)`).

There is also some logic within models (files inside `src/app/(models)`), they're mostly wrappers around DB queries.

And for E2E If this is possible for you I would much prefer playwright instead of Cypress.

And yes, please create an issue before making a PR 🙏. 

Other than that you have my go !

**Reply by Fredkiss3** [reply]
Also for components... I don't know if it possible yet to test server components within jest (or others), I mostly have seen people having issues about it.

So if it is too much of a task, or too complicated, please just make E2E tests, these can be enough.
